### PR TITLE
sql: add telemetry collection for enum usage

### LINKED ISSUE #53783: sql: add telemetry for enum usage
Add telemetry for when ENUM's are created and altered, and used in a table.

### PR DESCRIPTION
**Pull request by rohany**
Fixes #53783.

Release justification: low risk update to new functionality
Release note: None

### CODE REVIEW

File: pkg/sql/create_table.go [hunk 1]
```diff
@@ -1957,6 +1957,16 @@ func newTableDesc(
 		)
 	})
 
+	// See if any user defined types were used in this table.
+	if err := ret.ForeachPublicColumn(func(col *descpb.ColumnDescriptor) error {
```
Comment (otan) [inline pkg/sql/create_table.go:1961]: any reason to not re-use https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/create_table.go#L2205?

Comment (rohany) [reply pkg/sql/create_table.go:1961]: `ColumnDef`s contain only `tree.ResolveableTypeReferences`. We only know the type and everything after we process the def into a column

Comment (rohany) [reply pkg/sql/create_table.go:1961]: I'll just change it to operate on a column descriptor

**Review by otan — APPROVED** [review]

**Pull request merged**